crates/zksync/core/src/vm/farcall.rs

```diff
@@ -262,16 +181,16 @@ impl FarCallHandler {
                 Timestamp(state.local_state.timestamp),
             );
 
-            // change current stack to simulate return
-            let current = state.local_state.callstack.get_current_stack_mut();
-            current.pc = immediate_return.next_pc;
-            current.base_memory_page = MemoryPage(immediate_return.next_base_memory_page);
-            current.code_page = MemoryPage(immediate_return.next_code_page);
-            current.context_u128_value = immediate_return.next_context_u128_value;
-            current.sp = immediate_return.next_sp;
-            current.exception_handler_location = immediate_return.next_exception_handler_location;
-            current.this_address = immediate_return.next_this_address;
-            current.is_local_frame = immediate_return.next_is_local_frame;
+            // pop the current stack to simulate return
+            let new_cs = state.local_state.callstack.pop_entry();
```

Comment:
I guess this relies on the internal implementation of `RET` (kinda like before anyways), right?

Reply:
Yeah, but at least it's more intuitive as all implementations of `return` in almost all stacks pop the stack :)